### PR TITLE
fix(Core/playercreateinfo_item): Amount of allowed item

### PR DESCRIPTION
**Pull request by Kitzunu**
## CHANGES PROPOSED:
* Increase the amount allowed from tinyint signed to smallint unsigned

* Remove deprecated item from playercreateinfo_item

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3121
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/3119

## TESTS PERFORMED:
- Builds
- Tested to run Worldserver

## HOW TO TEST THE CHANGES:

1. Build
2. Run Worldserver, see so that you can add items > 127 on creation

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by Kitzunu** [comment]
Does this really need to be tested? :shrug:

**Comment by FrancescoBorzi** [comment]
@Kitzunu if you tested this, then I'd say it's small enough to be accepted.

File: data/sql/updates/pending_db_world/rev_1591992052509403400.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1591992052509403400');
+
+DELETE FROM `playercreateinfo_item` WHERE `itemid` = 40582 AND `amount` = -1;
+
+ALTER TABLE `playercreateinfo_item`
+	CHANGE `amount` `amount` SMALLINT UNSIGNED NOT NULL DEFAULT 1,
+	ADD `Note` VARCHAR(255) AFTER `amount`;
```
Comment (BarbzYHOOL) [inline data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: what is that `note` column about? @Kitzunu i feel like it's a mistake

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: So you dont have to google what the race and item are. So you can add a comment like "Warrior - Shadowedge"

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: call it "comment" then? like in other tables

and yes i can see it being useful, it won't take much space :D

**Comment by BarbzYHOOL** [comment]
if the CPP part looks fine to you @Helias @FrancescoBorzi then can be merged

File: data/sql/updates/pending_db_world/rev_1591992052509403400.sql (hunk 1 above)
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: I call it note because other playercreateinfo call it that

Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: hmm it's true on playercreateinfo_spell and spell_custom even though it's shit

dunno, mixed feelings

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: better do it this way and then refactor in the future imo

**Comment by FrancescoBorzi** [comment]
looks good, just a tiny note: changes to the DB **structure** are considered "Core" changes. With "DB" changes, we refer to changes to the **contents** of the DB, without altering its structure.

Just a convention. It has been like this since MaNGOS times ;)

File: data/sql/updates/pending_db_world/rev_1591992052509403400.sql (hunk 1 above)
Comment (BarbzYHOOL) [reply data/sql/updates/pending_db_world/rev_1591992052509403400.sql:7]: in fact the note is not used by the core so we could just rename to comment in this pr and also for the 2 other tables xD

**Comment by BarbzYHOOL** [comment]
> looks good, just a tiny note: changes to the DB **structure** are considered "Core" changes. With "DB" changes, we refer to changes to the **contents** of the DB, without altering its structure.
> 
> Just a convention. It has been like this since MaNGOS times ;)

good to know after 5 years doing the wrong thing then...

not sure if logic though since the db change is more consequent than the core change

File: src/server/game/Globals/ObjectMgr.cpp [hunk 2]
```diff
@@ -3333,7 +3333,7 @@ void ObjectMgr::LoadPlayerInfo()
                     continue;
                 }
 
-                int32 amount   = fields[3].GetInt8();
+                int32 amount   = fields[3].GetUInt16();
```
Comment (FrancescoBorzi) [inline src/server/game/Globals/ObjectMgr.cpp:3336]: @Kitzunu  I didn't notice this but why are we storing an unsigned int inside a signed `int32` ?

**Pull request merged**